-1

Tested at: 8e508d93df41869a855810a8c4fb9e3539e0f557

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26053/33374/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 64 COMPILER='llvm compile'

```


<pre>>> Compile sequence completed for CMSSW CMSSW_10_6_X_2019-03-03-0000
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
+ eval scram build outputlog '&&' '(/build/cmsbld/jenkins/workspace/ib-any-integration/cms-bot/buildLogAnalyzer.py' --logDir /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_6_X_2019-03-03-0000/tmp/slc7_amd64_gcc700/cache/log/src '||' 'true)'
++ scram build outputlog
>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_6_X_2019-03-03-0000/src/SimDataFormats/GeneratorProducts/test/rootio_t.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_6_X_2019-03-03-0000/src/SimDataFormats/GeneratorProducts/test/rootio_t.cc:21:22: error: constexpr variable 'kEnergy' must be initialized by a constant expression
    constexpr double kEnergy = sqrt( kMass*kMass+kMom*kMom);
                     ^         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_6_X_2019-03-03-0000/src/SimDataFormats/GeneratorProducts/test/rootio_t.cc:21:32: note: non-constexpr function 'sqrt' cannot be used in a constant expression
    constexpr double kEnergy = sqrt( kMass*kMass+kMom*kMom);
                               ^
</pre>

--- a/SimDataFormats/GeneratorProducts/src/hepmc_rootio.cc
+++ b/SimDataFormats/GeneratorProducts/src/hepmc_rootio.cc
@@ -2,9 +2,12 @@
 #include <HepMC/GenVertex.h>
 #include <iostream>
 
-//HACK
+//HACK We need to change the internals of GenVertex when reading
+// back via ROOT. We use the private access of GenEvent to 
+// accomplish it.
 namespace HepMC {
-  struct GenEvent {
+  class GenEvent {
+  public:
     static void clear_particles_in(HepMC::GenVertex* iVertex) {
       iVertex->m_particles_in.clear();
     }

--- a/SimDataFormats/GeneratorProducts/test/rootio_t.cc
+++ b/SimDataFormats/GeneratorProducts/test/rootio_t.cc
@@ -18,7 +18,7 @@ int main() {
 
     constexpr double kMass = 1.;
     constexpr double kMom = 1.;
-    constexpr double kEnergy = sqrt( kMass*kMass+kMom*kMom);
+    const double kEnergy = sqrt( kMass*kMass+kMom*kMom);
     constexpr int id = 11;
 
     auto p1 = new HepMC::GenParticle(HepMC::FourVector{kMom,0.,0.,kEnergy}, id, 1);

